py/server/tests/testbase.py

```diff
@@ -35,9 +35,11 @@ def tearDownClass(cls) -> None:
 
     def setUp(self) -> None:
         self._liveness_scope = liveness_scope()
+        self.opened_scope = liveness_scope().open()
```

Comment:
What we had before was too simple, and prevented an important use case: Keeping a LivenessScope around in order to fully control the life cycle of the objects created under it, and/or re-using the same LivenessScope on multiple calls. That is, the old API was simply not fit for purpose.

Reply:
This particular case is bad, because apparently the unittest "fixtures" can't be wrapped in contextmanagers or a `with` expr, etc - you can only explicitly open something, then close it later. That isn't a pattern we want to encourage here, since you might screw up and not close it. Apparently pytest does this better, closer to how junit solves this.

The other changes made in this patch's tests are more representative of how we expect this to work.